codex-rs/apply-patch/src/lib.rs

```diff
@@ -720,6 +724,65 @@ fn derive_new_contents_from_chunks(
     })
 }
 
+/// Return *only* the new file contents (joined into a single `String`) after
+/// applying the chunks to the file at `path`.
+fn derive_new_contents_from_chunks_ignoring_crlf(
+    path: &Path,
+    chunks: &[UpdateFileChunk],
+) -> std::result::Result<AppliedPatch, ApplyPatchError> {
+    let original_contents = match std::fs::read_to_string(path) {
+        Ok(contents) => contents,
+        Err(err) => {
+            return Err(ApplyPatchError::IoError(IoError {
+                context: format!("Failed to read file to update {}", path.display()),
+                source: err,
+            }));
+        }
+    };
+
+    // Split the original contents into lines, handling CRLF vs LF line endings appropriately.
+    let uses_crlf = original_contents_uses_crlf(&original_contents);
+    let original_lines: Vec<String> = original_contents.lines().map(String::from).collect();
+
+    let replacements = compute_replacements(&original_lines, path, chunks)?;
+    let new_lines = apply_replacements(original_lines, &replacements);
+    let mut new_lines = new_lines;
+    if !new_lines.last().is_some_and(String::is_empty) {
+        new_lines.push(String::new());
+    }
+    let new_contents = if uses_crlf {
+        new_lines.join("\r\n")
+    } else {
+        new_lines.join("\n")
+    };
+    Ok(AppliedPatch {
+        original_contents,
+        new_contents,
+    })
+}
+
+/// Detects whether the source file uses Windows CRLF line endings consistently.
+/// We only consider a file CRLF-formatted if every newline is part of a
+/// CRLF sequence. This avoids rewriting an LF-formatted file that merely
+/// contains embedded "\r\n" within string literals.
```

Comment:
Note that if you had the following string literal of JavaScript:

```javascript
"\r\n"
```

It contains neither `\n` nor `\r` characters, but the following six characters: double-quote, backslash, r, backslash, n, double-quote.

So I'm a bit confused by this comment.